components/lcd_touch/esp_lcd_touch/esp_lcd_touch.c

```diff
@@ -195,3 +199,27 @@ esp_err_t esp_lcd_touch_del(esp_lcd_touch_handle_t tp)
 
     return ESP_OK;
 }
+
+esp_err_t esp_lcd_touch_register_interrupt_callback(esp_lcd_touch_handle_t tp, esp_lcd_touch_interrupt_callback_t callback)
+{
+    esp_err_t ret = ESP_OK;
+    assert(tp != NULL);
+    assert(callback != NULL);
+
+    /* Interrupt pin is not selected */
+    if (tp->config.int_gpio_num == GPIO_NUM_NC) {
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    tp->config.interrupt_callback = callback;
+
+    /* Add GPIO ISR handler */
+    ret = gpio_install_isr_service(0);
+    ESP_RETURN_ON_ERROR(ret, TAG, "GPIO ISR install failed");
```

Comment:
This is a system-wide service. User might have already installed it, so `ESP_ERR_INVALID_STATE` should not be treated as error

Reply:
Good point, thank you. I fixed it.